Source/WTF/wtf/cocoa/UUIDCocoa.mm

```diff
@@ -35,11 +35,17 @@
     return [[NSUUID alloc] initWithUUIDString:toString()];
 }
 
-// Use string instead of bytes to avoid consideration of endianess (NSUUID stores UUID as array of bytes,
-// and we store UUID as UInt128).
-UUID::UUID(NSUUID * nsUUID)
-    : UUID(*parse(String([nsUUID UUIDString])))
+UUID::UUID(NSUUID *nsUUID)
+    : m_data(deletedValue)
 {
+    if (!nsUUID)
+        return;
+
+    // Use string instead of bytes to avoid consideration of endianess (NSUUID stores UUID as array of bytes,
+    // and we store UUID as UInt128).
+    auto uuid = parse(String([nsUUID UUIDString]));
+    ASSERT(uuid);
+    m_data = uuid->data();
```

Comment:
I am not sure we want to be able to construct a deleted or empty value without noticing it. Currently, we can create such UUID values explicitly only.
Would it be better to have a static std::optional<UUID> fromNSUUID method, similarly to the parse methods?

Reply:
Technically NSUUID with WebCore::UUID::deletedValue is a valid uuid, so we probably should make UUID accept this value (not treating it as invalid).

Reply:
As it is, this change could easily allow WebKit map corruptions.
If we want to support 1 as a valid UUID, we should remove the deleted UUID value and update our WebKit code accordingly as a first step.